src/components/NewDatePicker/index.js

```diff
@@ -0,0 +1,151 @@
+import React from 'react';
+import {Animated, View} from 'react-native';
+import moment from 'moment';
+import _ from 'underscore';
+import TextInput from '../TextInput';
+import CalendarPicker from '../CalendarPicker';
+import CONST from '../../CONST';
+import styles from '../../styles/styles';
+import {propTypes, defaultProps} from './datepickerPropTypes';
+import withWindowDimensions, {windowDimensionsPropTypes} from '../withWindowDimensions';
+import './styles.css';
+
+const datePickerPropTypes = {
+    ...propTypes,
+    ...windowDimensionsPropTypes,
+};
+
+const datePickerDefaultProps = {
+    ...defaultProps,
+};
+
+class NewDatePicker extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            selectedDate: new Date(),
+            isPickerVisible: false,
+        };
+
+        this.setDate = this.setDate.bind(this);
+        this.togglePicker = this.togglePicker.bind(this);
+        this.removeClickListener = this.removeClickListener.bind(this);
+        this.handleOutsideClick = this.handleOutsideClick.bind(this);
+
+        this.opacity = new Animated.Value(0);
+        this.wrapperRef = React.createRef();
+
+        /* We're using uncontrolled input otherwise it wont be possible to
+        * raise change events with a date value - each change will produce a date
+        * and make us reset the text input */
+        this.defaultValue = props.defaultValue
+            ? moment(props.defaultValue).format(CONST.DATE.MOMENT_FORMAT_STRING)
+            : '';
+    }
+
+    componentDidMount() {
+        document.addEventListener('mousedown', this.handleOutsideClick);
+    }
+
+    // eslint-disable-next-line rulesdir/prefer-early-return
+    componentDidUpdate(prevProps) {
+        if (prevProps.maxDate !== this.props.maxDate) {
+            document.addEventListener('mousedown', this.handleOutsideClick);
+        }
+    }
```

Comment:
Let's return early instead of disabling the rule
```suggestion
    // eslint-disable-next-line rulesdir/prefer-early-return
    componentDidUpdate(prevProps) {
        if (prevProps.maxDate === this.props.maxDate) {
            return;
        }
        document.addEventListener('mousedown', this.handleOutsideClick);
    }
```